sdk/core/core/inc/az_mut_span.h

```diff
@@ -0,0 +1,169 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#ifndef AZ_MUT_SPAN_H
+#define AZ_MUT_SPAN_H
+
+#include <az_span.h>
+#include <az_action.h>
+#include <az_contract.h>
+#include <az_result.h>
+#include <az_static_assert.h>
+
+#include <stdbool.h>
+#include <stdint.h>
+#include <string.h>
+
+#include <_az_cfg_prefix.h>
+
+/**
+ * A mutable span of bytes.
+ */
+AZ_SPAN_TYPE(az_mut_span, uint8_t)
+
+AZ_INLINE bool az_mut_span_is_empty(az_mut_span const span) { return span.size <= 0; }
+
+AZ_INLINE bool az_mut_span_is_valid(az_mut_span const span) {
+  return span.size == 0 || (span.begin != NULL && span.begin <= span.begin + span.size - 1);
+}
+
+/**
+ * Cast the given mutable span to an immutable span.
+ */
+AZ_INLINE az_span az_mut_span_to_span(az_mut_span const span) {
+  return (az_span){ .begin = span.begin, .size = span.size };
+}
+
+AZ_INLINE bool az_mut_span_is_overlap(az_mut_span const a, az_mut_span const b) {
+  return az_span_is_overlap(az_mut_span_to_span(a), az_mut_span_to_span(b));
+}
+
+AZ_INLINE az_result az_mut_span_copy(
+    az_mut_span const buffer,
+    az_span const src,
+    az_mut_span * const out_result) {
+  AZ_CONTRACT_ARG_NOT_NULL(out_result);
+  AZ_CONTRACT_ARG_VALID_SPAN(src);
+  AZ_CONTRACT_ARG_VALID_MUT_SPAN(buffer);
+
+  if (az_span_is_overlap(az_mut_span_to_span(buffer), src)) {
+    return AZ_ERROR_ARG;
+  }
+
+  if (buffer.size < src.size) {
+    return AZ_ERROR_BUFFER_OVERFLOW;
+  }
+
+  if (!az_span_is_empty(src)) {
+    memcpy((void *)buffer.begin, (void const *)src.begin, src.size);
```

Comment:
just do memmove

Reply:
https://github.com/Azure/azure-sdk-for-c/issues/128